src/registrar/forms/application_wizard.py

```diff
@@ -113,5 +116,34 @@ def get_context_data(self, form, **kwargs):
         context["form_titles"] = TITLES
         return context
 
-    def done(self, form_list, **kwargs):
+    def forms_to_object(self, form_dict: dict) -> DomainApplication:
+        """Unpack the form responses onto the model object properties."""
+        application = DomainApplication.objects.create(creator=self.request.user)
+
+        # organization information
+        organization_data = form_dict["organization"].cleaned_data
+        application.organization_type = organization_data["organization_type"]
+        application.federal_branch = organization_data["federal_type"]
+        application.is_election_office = organization_data["is_election_board"]
+
+        # contact information
+        contact_data = form_dict["contact"].cleaned_data
+        application.organization_name = contact_data["organization_name"]
+        application.street_address = contact_data["street_address"]
+        # TODO: add the rest of these fields when they are created in the forms
+
+        # This isn't really the requested_domain field
+        # but we need something in this field to make the form submittable
+        requested_site, _ = Website.objects.get_or_create(
+            website=contact_data["organization_name"] + ".gov"
+        )
```

Comment:
Your choice on whether you'd like to take care of it now or make a follow up ticket, but any data validation which is performed on the client side as a connivence to the user must be re-performed during saving. In other words, the user should get back a nice, readable message, not a random 500 error.

The steps to reproduce are 1. put "my domain name" in the organization name field; 2. click submit.